_posts/2022-01-20-saying-no.md

```diff
@@ -0,0 +1,43 @@
+
+To the software engineers reading this, I'm sure we've all worked with some APIs that have been great and some that have been terrible. An API with inconsistent and unpredictable behaviour with the same inputs is terrible. A poorly documented API is terrible. An API that attempts to accommodate your request and consistently times out rather than failing immediately is terrible.
+
+Humans are like APIs in this regard. A bad API for your coworkers is to be someone who gives radically different estimates for the same work based on whether you want to impress the person or not. A bad API for your coworkers is when you state you're on holiday or sick but still appear in conversations and meetings anyway. A bad API for your coworkers is when you tell them a 19:00 meeting is fine, but you're hangry and irritable because you've missed dinner and your children's bedtime.
+
+A previous performance of mine from [a great manager](https://github.com/derekprior) started with the sentence "Mike is very strict with his boundaries". "Oh no", I thought to myself, "I'm going to get criticised for always prioritising being there for my children's dinner and bedtimes four nights of the week". Instead, he went on to say that this made me easier to work with, set an excellent example for other members of the team to not overwork and set expectations for new parents on the team. Ironically, in this case, my being a little more selfish with my time and strict with my boundaries produced an outcome that benefitted the entire team.
+
+Having these boundaries clear and upfront is more complex but crucial for unexpected tasks. If you already feel overwhelmed or have too much to do when asked to do something new: you should say "no". If when asked for an estimate, [you can tell there's a bunch of complexity that was not considered by those who requested the estimate]({{ '/2021/11/29/hidden-complexities-of-software-estimation/' | absolute_url }}) and know that your realistic one will disappoint them: you should give them the realistic estimate.
+
+I call this "front-loading disappointment". You're better to have a conversation about your full workload or longer estimate now when people can prepare for it than making them happy now and then disappointing them in a few months when you didn't deliver what they planned for you. I've also found it to be less disruptive for your organisation for you to give an estimated deliverable earlier than planned than later than planned.
+
+### When to say "no"
+
+Ok, we've touched upon why you should say "no", so let's discuss when it's appropriate to do so.
+
+What's your current workload, the stress outside work and the number of commitments made to coworkers, family and friends look like now? What are your goals for the next 3-12 months and what's the best use of your time to achieve them? Regularly asking yourself these questions can help prime you to remind you of them when receiving a request.
```

Comment:
the lack of oxford comma actually tripped me up here, thinking that there were three things to look at:
1. current workload
2. the stress outside work and the number of commitments made to coworkers
3. family and friends look like now

Not sure, but this might clean it up
```suggestion
What's your current workload, the stress outside work, and the number of commitments made to coworkers, family, and friends look like now? What are your goals for the next 3-12 months and what's the best use of your time to achieve them? Regularly asking yourself these questions can help prime you to remind you of them when receiving a request.
```